helper.go

```diff
@@ -65,6 +65,19 @@ func imageExists(filename string) bool {
 		// means something wrong in exhaust file system
 		return false
 	}
+
+	// Check if there is lock in cache, retry after 1 second
+	if _, found := WriteLock.Get(filename); found {
```

Comment:
for(var i;i<=1;i++){
		// try to get lock...
		// break and return if necessary
	}